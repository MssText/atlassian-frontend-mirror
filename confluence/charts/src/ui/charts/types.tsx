export type ChartsProps = {
  /**
   * A `testId` prop is provided for specified elements, which is a unique
   * string that appears as a data attribute `data-testid` in the rendered code,
   * serving as a hook for automated tests */
  testId?: string;
  chartType: ChartTypes;
  data: any;
};

export enum ChartTypes {
  PIE = 'PIE',
  BAR = 'BAR',
  LINE = 'LINE',
}
