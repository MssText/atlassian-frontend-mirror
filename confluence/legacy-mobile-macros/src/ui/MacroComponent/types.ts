import type { ExtensionParams } from '@atlaskit/editor-common/extensions';

export interface MacroComponentProps {
  extension: ExtensionParams<any>;
  renderingStrategy?: string;
  createPromise: Function;
  eventDispatcher: any;
}
