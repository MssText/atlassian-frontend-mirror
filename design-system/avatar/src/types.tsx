import { AnalyticsEvent } from '@atlaskit/analytics-next';

export type AvatarSizeMap = Record<SizeType, number>;
export type AppearanceType = 'circle' | 'square';

export type SizeType =
  | 'xsmall'
  | 'small'
  | 'medium'
  | 'large'
  | 'xlarge'
  | 'xxlarge';

export type IndicatorSizeType = 'small' | 'medium' | 'large' | 'xlarge';

export type AvatarClickType = (
  event: React.MouseEvent,
  analyticsEvent?: AnalyticsEvent,
) => void;
