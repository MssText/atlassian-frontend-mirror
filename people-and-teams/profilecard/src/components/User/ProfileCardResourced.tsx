import React, { Suspense } from 'react';

import { GiveKudosLauncherLazy, KudosType } from '@atlaskit/give-kudos';

import { AnalyticsName } from '../../internal/analytics';
import filterActions from '../../internal/filterActions';
import { CardWrapper } from '../../styled/Card';
import {
  ProfileCardAction,
  ProfileCardClientData,
  ProfileCardResourcedProps,
  ProfileCardResourcedState,
  TeamCentralReportingLinesData,
} from '../../types';
import { ErrorMessage } from '../Error';

import ProfileCard from './ProfileCard';
import UserLoadingState from './UserLoadingState';

export default class ProfileCardResourced extends React.PureComponent<
  ProfileCardResourcedProps,
  ProfileCardResourcedState
> {
  static defaultProps: Partial<ProfileCardResourcedProps> = {
    actions: [],
  };

  _isMounted: boolean = false;

  state: ProfileCardResourcedState = {
    visible: false,
    isLoading: undefined,
    hasError: false,
    error: null,
    data: null,
    reportingLinesData: undefined,
    isKudosEnabled: false,
    kudosDrawerOpen: false,
  };

  componentDidMount() {
    this._isMounted = true;
    this.clientFetchProfile();
  }

  componentDidUpdate(
    prevProps: ProfileCardResourcedProps,
    prevState: ProfileCardResourcedState,
  ) {
    const { userId, cloudId, resourceClient } = this.props;
    const { hasError } = this.state;

    if (
      userId !== prevProps.userId ||
      cloudId !== prevProps.cloudId ||
      resourceClient !== prevProps.resourceClient
    ) {
      this.setState(
        {
          isLoading: undefined,
        },
        this.clientFetchProfile,
      );
    }

    if (hasError !== prevState.hasError && hasError) {
      this.callAnalytics(AnalyticsName.PROFILE_CARD_RESOURCED_ERROR);
    }
  }

  componentWillUnmount() {
    this._isMounted = false;
  }

  private callAnalytics = (id: string, options: any = {}) => {
    const { analytics } = this.props;
    if (analytics) {
      analytics(id, options);
    }
  };

  clientFetchProfile = () => {
    const { cloudId, userId } = this.props;
    const { isLoading } = this.state;

    if (isLoading === true) {
      // don't fetch data when fetching is in process
      return;
    }

    this.setState(
      {
        isLoading: true,
        hasError: false,
        data: null,
      },
      () => {
        const requests = Promise.all([
          this.props.resourceClient.getProfile(cloudId, userId),
          this.props.resourceClient.getReportingLines(userId),
          this.props.resourceClient.shouldShowGiveKudos(),
        ]);

        requests
          .then(
            (res) => this.handleClientSuccess(...res),
            (err) => this.handleClientError(err),
          )
          .catch((err) => this.handleClientError(err));
      },
    );
  };

  handleClientSuccess(
    profileData: ProfileCardClientData,
    reportingLinesData: TeamCentralReportingLinesData,
    shouldShowGiveKudos: boolean,
  ) {
    if (!this._isMounted) {
      return;
    }

    this.setState({
      isLoading: false,
      hasError: false,
      data: profileData,
      reportingLinesData,
      isKudosEnabled: shouldShowGiveKudos,
    });
  }

  handleClientError(err: any) {
    if (!this._isMounted) {
      return;
    }

    this.setState({
      isLoading: false,
      hasError: true,
      error: err,
    });
  }

  filterActions = (): ProfileCardAction[] =>
    filterActions(this.props.actions, this.state.data);

  openKudosDrawer = () => {
    this.setState({ kudosDrawerOpen: true });
  };

  closeKudosDrawer = () => {
    this.setState({ kudosDrawerOpen: false });
  };

  render(): React.ReactNode {
    const {
      isLoading,
      hasError,
      error,
      data,
      reportingLinesData,
      isKudosEnabled,
    } = this.state;
    const {
      analytics,
      onReportingLinesClick,
      cloudId,
      userId,
      addFlag,
    } = this.props;

    const isFetchingOrNotStartToFetchYet =
      isLoading === true || isLoading === undefined;

    if (isFetchingOrNotStartToFetchYet) {
      return (
        <CardWrapper>
          <UserLoadingState />
        </CardWrapper>
      );
    } else if (hasError) {
      return (
        <CardWrapper>
          <ErrorMessage errorType={error} reload={this.clientFetchProfile} />
        </CardWrapper>
      );
    }

    const newProps = {
      hasError,
      errorType: error,
      clientFetchProfile: this.clientFetchProfile,
      analytics,
      reportingLines: reportingLinesData,
      onReportingLinesClick: onReportingLinesClick,
      cloudId,
      userId,
      addFlag,
      ...data,
      isKudosEnabled,
      teamCentralBaseUrl: this.props.resourceClient.getTeamCentralBaseUrl(),
      openKudosDrawer: this.openKudosDrawer,
    };

    return (
      <CardWrapper>
        <>
          {isKudosEnabled && (
            <Suspense fallback={null}>
              <GiveKudosLauncherLazy
                isOpen={this.state.kudosDrawerOpen}
                recipient={{
                  type: KudosType.INDIVIDUAL,
                  recipientId: this.props.userId!,
                }}
                analyticsSource="profile-card"
                teamCentralBaseUrl={newProps.teamCentralBaseUrl!}
                cloudId={this.props.cloudId!}
                addFlag={this.props.addFlag}
                onClose={this.closeKudosDrawer}
              />
            </Suspense>
          )}
          <ProfileCard {...newProps} actions={this.filterActions()} />
        </>
      </CardWrapper>
    );
  }
}
