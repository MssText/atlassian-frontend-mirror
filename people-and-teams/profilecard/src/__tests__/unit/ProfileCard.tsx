import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from '@atlaskit/button/custom-theme-button';
import ProfileCardResourced, { ProfileCard, ProfileClient } from '../..';
import ErrorMessage from '../../components/ErrorMessage';

import { ProfilecardProps } from '../../types';

import {
  FullNameLabel,
  ActionButtonGroup,
  SpinnerContainer,
} from '../../styled/Card';
import mockGlobalDate from './helper/_mock-global-date';

describe('Profilecard', () => {
  const defaultProps: ProfilecardProps = {
    fullName: 'full name test',
    status: 'active',
    nickname: 'jscrazy',
    companyName: 'Atlassian',
  };

  const TODAY = new Date(2018, 10, 19, 17, 30, 0, 0);

  beforeAll(() => {
    mockGlobalDate.setToday(TODAY);
  });

  afterAll(() => {
    mockGlobalDate.reset();
  });

  const renderShallow = (props = {}) =>
    shallow(<ProfileCard {...defaultProps} {...props} />);

  it('should export default ProfileCardResourced', () => {
    expect(ProfileCardResourced).toBeInstanceOf(Object);
  });

  it('should export named ProfileCard and ProfileClient', () => {
    expect(ProfileCard).toBeInstanceOf(Object);
    expect(ProfileClient).toBeInstanceOf(Object);
  });

  describe('ProfileCard', () => {
    it('should be possible to create a component', () => {
      const card = shallow(<ProfileCard />);
      expect(card.length).toBeGreaterThan(0);
    });

    describe('fullName property', () => {
      const fullName = 'This is an avatar!';
      const card = shallow(<ProfileCard fullName={fullName} />);

      it('should show the full name on the card if property is set', () => {
        const el = card.find(FullNameLabel).dive();
        expect(el.text()).toBe(fullName);
      });

      it('should not render a card if full name is not set', () => {
        card.setProps({ fullName: undefined });
        expect(card.find(ProfileCard).children()).toHaveLength(0);
      });

      it('should match snapshot when fullName and nickName are equal', () => {
        const wrapper = renderShallow({
          fullName: 'Same Same',
          nickname: 'Same Same',
        });
        expect(wrapper).toMatchSnapshot();
      });

      it('should match snapshot when fullName and nickName are set', () => {
        const wrapper = renderShallow();
        expect(wrapper).toMatchSnapshot();
      });

      it('should match snapshot when nickName is missing', () => {
        const wrapper = renderShallow({ nickname: undefined });
        expect(wrapper).toMatchSnapshot();
      });
    });

    describe('isLoading property', () => {
      it('should render the LoadingMessage component', () => {
        const card = shallow(<ProfileCard isLoading />);
        expect(card.find(SpinnerContainer).exists()).toBe(true);
      });
    });

    describe('hasError property', () => {
      it('should render the ErrorMessage component', () => {
        const card = shallow(<ProfileCard hasError />);
        expect(card.find(ErrorMessage).exists()).toBe(true);
      });

      it('should render the ErrorMessage component with retry button if clientFetchProfile is provided', () => {
        const card = mount(<ProfileCard hasError />);

        const errorComponent = card.find(ErrorMessage);
        expect(errorComponent.length).toBe(1);
        // expect(errorComponent.find(CrossCircleIcon).length).toBe(1);
        expect(errorComponent.find(Button).length).toBe(1);
      });
    });

    describe('actions property', () => {
      const actions = [
        {
          id: 'one',
          label: 'one',
        },
        {
          id: 'two',
          label: 'two',
        },
        {
          id: 'three',
          label: 'three',
        },
      ];
      const card = mount(<ProfileCard fullName="name" actions={actions} />);

      it('should render an action button for every item in actions property', () => {
        const actionsWrapper = card.find(ActionButtonGroup);
        const buttonTexts = card
          .find(Button)
          .children()
          .map(node => node.text());

        expect(actionsWrapper.children().first().children()).toHaveLength(
          actions.length,
        );
        expect(buttonTexts).toEqual(actions.map(action => action.label));
      });

      it('should call callback handler when action button is clicked', () => {
        const spy = jest.fn().mockImplementation(() => {}); // eslint-disable-line no-undef
        card.setProps({
          actions: [
            {
              label: 'test',
              callback: spy,
            },
          ],
        });
        const actionsWrapper = card.find(ActionButtonGroup);
        actionsWrapper.find(Button).first().simulate('click');
        expect(spy.mock.calls.length).toBe(1);
      });

      it('should not render any action buttons if actions property is not set', () => {
        card.setProps({ actions: undefined });
        const actionsWrapper = card.find(ActionButtonGroup);
        expect(actionsWrapper.children().length).toBe(0);
      });

      it('link default behaviour should be prevented if a callback is provided', () => {
        const spy = jest.fn().mockImplementation(() => {});
        const preventDefault = jest.fn().mockImplementation(() => {});
        card.setProps({
          actions: [
            {
              label: 'test',
              callback: spy,
              link: '#',
            },
          ],
        });
        const actionButton = card.find(ActionButtonGroup).find(Button).first();
        expect(actionButton.prop('href')).toBe('#');
        actionButton.simulate('click', { preventDefault });
        expect(spy.mock.calls.length).toBe(1);
        expect(preventDefault.mock.calls.length).toBe(1);
      });

      it('link default behaviour should not be prevented if no callback provided', () => {
        const preventDefault = jest.fn().mockImplementation(() => {});
        card.setProps({
          actions: [
            {
              label: 'test',
              link: '#',
            },
          ],
        });
        const actionButton = card.find(ActionButtonGroup).find(Button).first();
        expect(actionButton.prop('href')).toBe('#');
        actionButton.simulate('click', { preventDefault });
        expect(preventDefault.mock.calls.length).toBe(0);
      });
    });

    describe('status property', () => {
      it('should match snapshot when status=inactive and status modified date is unknown', () => {
        const card = renderShallow({
          status: 'inactive',
          statusModifiedDate: undefined,
        });

        expect(card).toMatchSnapshot();
      });

      it('should match snapshot when status=inactive and status modified date is defined', () => {
        const card = renderShallow({
          status: 'inactive',
          statusModifiedDate: 1542608651819,
        });

        expect(card).toMatchSnapshot();
      });

      it('should match snapshot when status=closed and status modified date is unknown', () => {
        const card = renderShallow({
          fullName: undefined,
          status: 'closed',
          statusModifiedDate: undefined,
        });

        expect(card).toMatchSnapshot();
      });

      it('should match snapshot when status=closed and status modified date is defined', () => {
        const card = renderShallow({
          fullName: undefined,
          status: 'closed',
          statusModifiedDate: 1542608651819,
        });

        expect(card).toMatchSnapshot();
      });

      it('should match snapshot when status=closed and hasDisabledAccountLozenge=false', () => {
        const card = renderShallow({
          status: 'closed',
          hasDisabledAccountLozenge: false,
        });

        expect(card).toMatchSnapshot();
      });

      it('should match snapshot when status=inactive and hasDisabledAccountLozenge=false', () => {
        const card = renderShallow({
          status: 'inactive',
          hasDisabledAccountLozenge: false,
        });

        expect(card).toMatchSnapshot();
      });

      it('should match snapshot when status=closed and disabledAccountMessage is defined', () => {
        const card = renderShallow({
          status: 'closed',
          disabledAccountMessage: (
            <p>this is a custom message for closed account</p>
          ),
        });

        expect(card).toMatchSnapshot();
      });

      it('should match snapshot when status=inactive and disabledAccountMessage is defined', () => {
        const card = renderShallow({
          status: 'inactive',
          disabledAccountMessage: (
            <p>this is a custom message for inactive account</p>
          ),
        });

        expect(card).toMatchSnapshot();
      });
    });
  });
});
