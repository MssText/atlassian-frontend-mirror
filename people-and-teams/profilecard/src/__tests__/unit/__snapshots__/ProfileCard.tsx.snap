// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profilecard ProfileCard customLozenges property should match snapshot when multiple custom lozenges are specified 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={false}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          noMeta={true}
        >
          full name test (jscrazy) 
        </styled.span>
        <Styled(styled.div)>
          <Lozenge
            appearance="new"
            isBold={true}
            key="0"
          >
            Guest
          </Lozenge>
          <Lozenge
            appearance="removed"
            key="1"
          >
            Cool Bean
          </Lozenge>
          <Lozenge
            appearance="inprogress"
            isBold={true}
            key="2"
          >
            <div>
              Another Role
            </div>
          </Lozenge>
        </Styled(styled.div)>
        <IconLabel
          icon="email"
        />
        <IconLabel
          icon="time"
        />
        <IconLabel
          icon="companyName"
        >
          Atlassian
        </IconLabel>
        <IconLabel
          icon="location"
        />
        <ReportingLinesDetails
          analytics={[Function]}
          getDuration={[Function]}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard customLozenges property should match snapshot when no custom lozenges are specified 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={false}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          noMeta={true}
        >
          full name test (jscrazy) 
        </styled.span>
        <IconLabel
          icon="email"
        />
        <IconLabel
          icon="time"
        />
        <IconLabel
          icon="companyName"
        >
          Atlassian
        </IconLabel>
        <IconLabel
          icon="location"
        />
        <ReportingLinesDetails
          analytics={[Function]}
          getDuration={[Function]}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard customLozenges property should match snapshot when one custom lozenge is specified 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={false}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          noMeta={true}
        >
          full name test (jscrazy) 
        </styled.span>
        <Styled(styled.div)>
          <Lozenge
            appearance="new"
            isBold={true}
            key="0"
          >
            Guest
          </Lozenge>
        </Styled(styled.div)>
        <IconLabel
          icon="email"
        />
        <IconLabel
          icon="time"
        />
        <IconLabel
          icon="companyName"
        >
          Atlassian
        </IconLabel>
        <IconLabel
          icon="location"
        />
        <ReportingLinesDetails
          analytics={[Function]}
          getDuration={[Function]}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard fullName property should match snapshot when fullName and nickName are equal 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={false}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          noMeta={true}
        >
          Same Same
        </styled.span>
        <IconLabel
          icon="email"
        />
        <IconLabel
          icon="time"
        />
        <IconLabel
          icon="companyName"
        >
          Atlassian
        </IconLabel>
        <IconLabel
          icon="location"
        />
        <ReportingLinesDetails
          analytics={[Function]}
          getDuration={[Function]}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard fullName property should match snapshot when fullName and nickName are set 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={false}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          noMeta={true}
        >
          full name test (jscrazy) 
        </styled.span>
        <IconLabel
          icon="email"
        />
        <IconLabel
          icon="time"
        />
        <IconLabel
          icon="companyName"
        >
          Atlassian
        </IconLabel>
        <IconLabel
          icon="location"
        />
        <ReportingLinesDetails
          analytics={[Function]}
          getDuration={[Function]}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard fullName property should match snapshot when nickName is missing 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={false}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          noMeta={true}
        >
          full name test
        </styled.span>
        <IconLabel
          icon="email"
        />
        <IconLabel
          icon="time"
        />
        <IconLabel
          icon="companyName"
        >
          Atlassian
        </IconLabel>
        <IconLabel
          icon="location"
        />
        <ReportingLinesDetails
          analytics={[Function]}
          getDuration={[Function]}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard reportingLines property should match snapshot when no reporting lines are specified 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={false}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          noMeta={true}
        >
          full name test (jscrazy) 
        </styled.span>
        <IconLabel
          icon="email"
        />
        <IconLabel
          icon="time"
        />
        <IconLabel
          icon="companyName"
        >
          Atlassian
        </IconLabel>
        <IconLabel
          icon="location"
        />
        <ReportingLinesDetails
          analytics={[Function]}
          getDuration={[Function]}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard reportingLines property should match snapshot when no reporting lines are specified 2`] = `
<styled.div>
  <styled.div
    isDisabledUser={false}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          noMeta={true}
        >
          full name test (jscrazy) 
        </styled.span>
        <IconLabel
          icon="email"
        />
        <IconLabel
          icon="time"
        />
        <IconLabel
          icon="companyName"
        >
          Atlassian
        </IconLabel>
        <IconLabel
          icon="location"
        />
        <ReportingLinesDetails
          analytics={[Function]}
          getDuration={[Function]}
          reportingLines={Object {}}
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard reportingLines property should match snapshot when no reporting lines are specified 3`] = `
<styled.div>
  <styled.div
    isDisabledUser={false}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          noMeta={true}
        >
          full name test (jscrazy) 
        </styled.span>
        <IconLabel
          icon="email"
        />
        <IconLabel
          icon="time"
        />
        <IconLabel
          icon="companyName"
        >
          Atlassian
        </IconLabel>
        <IconLabel
          icon="location"
        />
        <ReportingLinesDetails
          analytics={[Function]}
          getDuration={[Function]}
          reportingLines={
            Object {
              "managers": Array [],
              "reports": Array [],
            }
          }
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard reportingLines property should match snapshot when reporting lines are specified 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={false}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          noMeta={true}
        >
          full name test (jscrazy) 
        </styled.span>
        <IconLabel
          icon="email"
        />
        <IconLabel
          icon="time"
        />
        <IconLabel
          icon="companyName"
        >
          Atlassian
        </IconLabel>
        <IconLabel
          icon="location"
        />
        <ReportingLinesDetails
          analytics={[Function]}
          getDuration={[Function]}
          reportingLines={
            Object {
              "managers": Array [
                Object {
                  "accountIdentifier": "abcd",
                  "identifierType": "ATLASSIAN_ID",
                  "pii": Object {
                    "name": "name",
                    "picture": "picture",
                  },
                },
              ],
              "reports": Array [
                Object {
                  "accountIdentifier": "abcd",
                  "identifierType": "ATLASSIAN_ID",
                  "pii": Object {
                    "name": "name",
                    "picture": "picture",
                  },
                },
                Object {
                  "accountIdentifier": "abcd",
                  "identifierType": "ATLASSIAN_ID",
                  "pii": Object {
                    "name": "name",
                    "picture": "picture",
                  },
                },
              ],
            }
          }
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard status property should match snapshot when status=closed and disabledAccountMessage is defined 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={true}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          isDisabledAccount={true}
          noMeta={true}
        >
          jscrazy
        </styled.span>
        <styled.div>
          <Lozenge
            appearance="default"
            isBold={true}
          >
            <MemoizedFormattedMessage
              defaultMessage="Account deleted"
              description="A text in a grey lozenge shows that this user is closed/deleted"
              id="pt.profile-card.closed.account"
            />
          </Lozenge>
        </styled.div>
        <styled.div>
          <p>
            this is a custom message for closed account
          </p>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard status property should match snapshot when status=closed and hasDisabledAccountLozenge=false 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={true}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          isDisabledAccount={true}
          noMeta={true}
        >
          jscrazy
        </styled.span>
        <styled.div>
          <p>
            <MemoizedFormattedMessage
              defaultMessage="You can no longer collaborate with this person."
              description="A first sentence of a long text explains this user is inactive/closed"
              id="pt.profile-card.general.msg.disabled.user"
            />
             
            <MemoizedFormattedMessage
              defaultMessage="Their account has been deleted."
              description="A long text explains this user is closed when we do not know the date of starting to close"
              id="pt.profile-card.closed.account.no.date"
            />
          </p>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard status property should match snapshot when status=closed and status modified date is defined 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={true}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          isDisabledAccount={true}
          noMeta={true}
        >
          jscrazy
        </styled.span>
        <styled.div>
          <Lozenge
            appearance="default"
            isBold={true}
          >
            <MemoizedFormattedMessage
              defaultMessage="Account deleted"
              description="A text in a grey lozenge shows that this user is closed/deleted"
              id="pt.profile-card.closed.account"
            />
          </Lozenge>
        </styled.div>
        <styled.div>
          <p>
            <MemoizedFormattedMessage
              defaultMessage="You can no longer collaborate with this person."
              description="A first sentence of a long text explains this user is inactive/closed"
              id="pt.profile-card.general.msg.disabled.user"
            />
             
            <MemoizedFormattedMessage
              defaultMessage="Their account has been deleted."
              description="A long text explains this user is closed when we do not know the date of starting to close"
              id="pt.profile-card.closed.account.no.date"
            />
          </p>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard status property should match snapshot when status=closed and status modified date is unknown 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={true}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          isDisabledAccount={true}
          noMeta={true}
        >
          jscrazy
        </styled.span>
        <styled.div>
          <Lozenge
            appearance="default"
            isBold={true}
          >
            <MemoizedFormattedMessage
              defaultMessage="Account deleted"
              description="A text in a grey lozenge shows that this user is closed/deleted"
              id="pt.profile-card.closed.account"
            />
          </Lozenge>
        </styled.div>
        <styled.div>
          <p>
            <MemoizedFormattedMessage
              defaultMessage="You can no longer collaborate with this person."
              description="A first sentence of a long text explains this user is inactive/closed"
              id="pt.profile-card.general.msg.disabled.user"
            />
             
            <MemoizedFormattedMessage
              defaultMessage="Their account has been deleted."
              description="A long text explains this user is closed when we do not know the date of starting to close"
              id="pt.profile-card.closed.account.no.date"
            />
          </p>
        </styled.div>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard status property should match snapshot when status=inactive and disabledAccountMessage is defined 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={true}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          isDisabledAccount={true}
          noMeta={true}
        >
          full name test
        </styled.span>
        <styled.div>
          <Lozenge
            appearance="default"
            isBold={true}
          >
            <MemoizedFormattedMessage
              defaultMessage="Account deactivated"
              description="A text in a grey lozenge shows that this user is inactive"
              id="pt.profile-card.inactive.account"
            />
          </Lozenge>
        </styled.div>
        <styled.div>
          <p>
            this is a custom message for inactive account
          </p>
        </styled.div>
        <IconLabel
          icon="companyName"
        >
          Atlassian
        </IconLabel>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard status property should match snapshot when status=inactive and hasDisabledAccountLozenge=false 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={true}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          isDisabledAccount={true}
          noMeta={true}
        >
          full name test
        </styled.span>
        <styled.div>
          <p>
            <MemoizedFormattedMessage
              defaultMessage="You can no longer collaborate with this person."
              description="A first sentence of a long text explains this user is inactive/closed"
              id="pt.profile-card.general.msg.disabled.user"
            />
             
            <MemoizedFormattedMessage
              defaultMessage="Their account has been deactivated."
              description="A long text explains this user is inactive when we do not know the date of starting to deactivate"
              id="pt.profile-card.inactive.account.no.date"
            />
          </p>
        </styled.div>
        <IconLabel
          icon="companyName"
        >
          Atlassian
        </IconLabel>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard status property should match snapshot when status=inactive and status modified date is defined 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={true}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          isDisabledAccount={true}
          noMeta={true}
        >
          full name test
        </styled.span>
        <styled.div>
          <Lozenge
            appearance="default"
            isBold={true}
          >
            <MemoizedFormattedMessage
              defaultMessage="Account deactivated"
              description="A text in a grey lozenge shows that this user is inactive"
              id="pt.profile-card.inactive.account"
            />
          </Lozenge>
        </styled.div>
        <styled.div>
          <p>
            <MemoizedFormattedMessage
              defaultMessage="You can no longer collaborate with this person."
              description="A first sentence of a long text explains this user is inactive/closed"
              id="pt.profile-card.general.msg.disabled.user"
            />
             
            <MemoizedFormattedMessage
              defaultMessage="Their account has been deactivated."
              description="A long text explains this user is inactive when we do not know the date of starting to deactivate"
              id="pt.profile-card.inactive.account.no.date"
            />
          </p>
        </styled.div>
        <IconLabel
          icon="companyName"
        >
          Atlassian
        </IconLabel>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`Profilecard ProfileCard status property should match snapshot when status=inactive and status modified date is unknown 1`] = `
<styled.div>
  <styled.div
    isDisabledUser={true}
  >
    <styled.div>
      <Avatar
        borderColor="var(--ds-surface-overlay, #FFFFFF)"
        size="xlarge"
      />
    </styled.div>
    <styled.div>
      <styled.div>
        <styled.span
          isDisabledAccount={true}
          noMeta={true}
        >
          full name test
        </styled.span>
        <styled.div>
          <Lozenge
            appearance="default"
            isBold={true}
          >
            <MemoizedFormattedMessage
              defaultMessage="Account deactivated"
              description="A text in a grey lozenge shows that this user is inactive"
              id="pt.profile-card.inactive.account"
            />
          </Lozenge>
        </styled.div>
        <styled.div>
          <p>
            <MemoizedFormattedMessage
              defaultMessage="You can no longer collaborate with this person."
              description="A first sentence of a long text explains this user is inactive/closed"
              id="pt.profile-card.general.msg.disabled.user"
            />
             
            <MemoizedFormattedMessage
              defaultMessage="Their account has been deactivated."
              description="A long text explains this user is inactive when we do not know the date of starting to deactivate"
              id="pt.profile-card.inactive.account.no.date"
            />
          </p>
        </styled.div>
        <IconLabel
          icon="companyName"
        >
          Atlassian
        </IconLabel>
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
